code/scripting/api/objs/oswpt.cpp

```diff
@@ -0,0 +1,183 @@
+//
+//
+
+#include "oswpt.h"
+#include "parse_object.h"
+#include "ship.h"
+#include "team.h"
+#include "waypoint.h"
+#include "wing.h"
+#include "ship/ship.h"
+
+namespace scripting {
+namespace api {
+
+//**********HANDLE: Wing
+ADE_OBJ(l_OSWPT, object_ship_wing_point_team, "oswpt", "OSWPT handle");
+
+ADE_FUNC(getType, l_OSWPT, nullptr, "The data-type this OSWPT yields on the get method.", "string", "The name of the data type. Either 'ship', 'parseobject' (a yet-to-spawn ship), 'wing' (can include yet-to-arrive wings with 0 current ships), 'team' (both explicit and ship-on-team), 'waypoint',  or 'none' (either explicitly specified, a ship that doesn't exist anymore, or invalid OSWPT object).")
+{
+	object_ship_wing_point_team oswpt;
+	if (!ade_get_args(L, "o", l_OSWPT.Get(&oswpt)))
+		return ade_set_error(L, "s", "none");
+
+	switch (oswpt.type) {
+	case OSWPT_TYPE_SHIP:
+		return ade_set_args(L, "s", "ship");
+	case OSWPT_TYPE_PARSE_OBJECT:
+		return ade_set_args(L, "s", "parseobject");
+	case OSWPT_TYPE_WING:
+	case OSWPT_TYPE_WING_NOT_PRESENT:
+		return ade_set_args(L, "s", "wing");
+	case OSWPT_TYPE_SHIP_ON_TEAM:
```

Comment:
`SHIP_ON_TEAM` should be distinguished separately

Reply:
Since this is about the data-type that the scripter would have to deal with (which can vary for multiple invocations of the sexp), whereas the ships-on-team vs team distiction is always the same for the whole sexp, it makes sense to keep them handled the same here